test/unit/ioc.spec.js

```diff
@@ -142,6 +142,44 @@ test.group('Ioc', function () {
     assert.equal(ioc.getAutoloads()['App'], path.join(__dirname, './app'))
   })
 
+  test('should be able to call a closure when loading has been successful using with', function () {
+    const ioc = new Ioc()
+    const fooFn = function () {
+      return 'foo'
+    }
+    ioc.bind('App/Foo', fooFn)
+
+    ioc.with('App/Foo', (Foo) => {
+      assert.equal(Foo, 'foo')
```

Comment:
Where do you get the `assert` from? 

I suggest planning for assertions to avoid false negatives